pkg/configmap/configmap_watcher.go

```diff
@@ -31,55 +33,99 @@ type KubeLinterChecks struct {
+func NewWatcher(cfg *rest.Config) (Watcher, error) {
 	clientset, err := kubernetes.NewForConfig(cfg)
 	if err != nil {
-		return ConfigMapWatcher{}, fmt.Errorf("initializing clientset: %w", err)
+		return Watcher{}, fmt.Errorf("initializing clientset: %w", err)
 	}
 
-	return ConfigMapWatcher{
+	// the Informer will use this to monitor the namespace for the ConfigMap.
+	namespace, err := getPodNamespace()
+	if err != nil {
+		return Watcher{}, fmt.Errorf("getting namespace: %w", err)
+	}
+
+	return Watcher{
 		clientset: clientset,
 		logger:    log.Log.WithName("ConfigMapWatcher"),
+		ch:        make(chan config.Config),
+		namespace: namespace,
 	}, nil
 }
 
 // GetStaticKubelinterConfig returns the ConfigMap's checks configuration
-func (cmw *ConfigMapWatcher) GetStaticKubelinterConfig(ctx context.Context) (config.Config, error) {
+func (cmw *Watcher) GetStaticKubelinterConfig(ctx context.Context) (config.Config, error) {
 	cm, err := cmw.clientset.CoreV1().
-		ConfigMaps(configMapNamespace).Get(ctx, configMapName, v1.GetOptions{})
+		ConfigMaps(cmw.namespace).Get(ctx, configMapName, v1.GetOptions{})
 	if err != nil {
 		return config.Config{}, fmt.Errorf("getting initial configuration: %w", err)
 	}
 
 	return cmw.getKubeLinterConfig(cm.Data[configMapDataAccess])
 }
 
-// StartInformer will update the channel structure with new configuration data from ConfigMap update event
-func (cmw *ConfigMapWatcher) StartInformer(ctx context.Context) error {
+// Start will update the channel structure with new configuration data from ConfigMap update event
+func (cmw Watcher) Start(ctx context.Context) error {
 	factory := informers.NewSharedInformerFactoryWithOptions(
-		cmw.clientset, time.Second*30, informers.WithNamespace(configMapNamespace),
+		cmw.clientset, time.Second*30, informers.WithNamespace(cmw.namespace),
 	)
 	informer := factory.Core().V1().ConfigMaps().Informer()
 
 	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{ // nolint:errcheck
+		AddFunc: func(obj interface{}) {
+			newCm := obj.(*apicorev1.ConfigMap)
+
+			if configMapName != newCm.GetName() {
+				return
+			}
+
+			cmw.logger.Info(
+				"a ConfigMap has been created under watched namespace",
+				"name", newCm.GetName(),
+				"namespace", newCm.GetNamespace(),
+			)
+
+			cfg, err := cmw.getKubeLinterConfig(newCm.Data[configMapDataAccess])
+			if err != nil {
+				cmw.logger.Error(err, "ConfigMap data format")
+				return
+			}
+
+			cmw.ch <- cfg
+		},
 		UpdateFunc: func(oldObj, newObj interface{}) {
 			newCm := newObj.(*apicorev1.ConfigMap)
 
-			cmw.logger.Info("ConfigMap has been updated")
+			// This is sometimes triggered even if no change was due to the ConfigMap
+			if configMapName != newCm.GetName() || reflect.DeepEqual(oldObj, newObj) {
+				return
+			}
+
+			cmw.logger.Info(
+				"a ConfigMap has been updated under watched namespace",
+				"name", newCm.GetName(),
+				"namespace", newCm.GetNamespace(),
+			)
```

Comment:
Probably an edge case, but we also need a DeleteFunc (when the configmap is removed then return back to the default conf). I think the logic for all the funcions will be basically the same so perhaps it would be good to export to a function and reuse.